src/control-plane/services.ts

```diff
@@ -65,18 +62,29 @@ export class Services extends Construct {
       true // applyToChildren = true, so that it applies to policies created for the role.
     );
 
+    // https://docs.powertools.aws.dev/lambda/python/2.31.0/#lambda-layer
+    const lambdaPowerToolsLayerARN = `arn:aws:lambda:${
+      Stack.of(this).region
+    }:017000801446:layer:AWSLambdaPowertoolsPythonV2:59`;
+
     const tenantManagementServices = new PythonFunction(this, 'TenantManagementServices', {
-      entry: path.join(__dirname, '../../resources/functions/'),
+      entry: path.join(__dirname, '../../resources/functions/tenant-management'),
       runtime: Runtime.PYTHON_3_12,
       index: 'tenant_management.py',
       handler: 'lambda_handler',
       timeout: Duration.seconds(60),
       role: tenantManagementExecRole,
-      layers: [props.lambdaLayer],
+      layers: [
+        LayerVersion.fromLayerVersionArn(this, 'LambdaPowerTools', lambdaPowerToolsLayerARN),
+      ],
       environment: {
```

Comment:
Assuming you export the enum, i.e.: 

```typescript
export enum EventType {
  ONBOARDING_REQUEST = 'onboardingRequest',
  OFFBOARDING_REQUEST = 'offboardingRequest',
  ACTIVATE_REQUEST = 'activateRequest',
  DEACTIVATE_REQUEST = 'deactivateRequest',
}
```
Then this code can just be:

```typescript
      environment: {
        EVENTBUS_NAME: props.eventManager.eventBus.eventBusName,
        EVENT_SOURCE: props.eventManager.controlPlaneEventSource,
        TENANT_DETAILS_TABLE: props.tables.tenantDetails.tableName,
        ONBOARDING_DETAIL_TYPE: EventType.ONBOARDING_REQUEST,
        OFFBOARDING_DETAIL_TYPE: EventType.OFFBOARDING_REQUEST,
        ACTIVATE_DETAIL_TYPE: EventType.ACTIVATE_REQUEST,
        DEACTIVATE_DETAIL_TYPE: EventType.DEACTIVATE_REQUEST,
      },
```

Reply:
Updated to use enum as detailType.